sycl/include/CL/sycl/group.hpp

```diff
@@ -397,6 +411,99 @@ template <int Dimensions = 1> class group {
   range<Dimensions> groupRange;
   id<Dimensions> index;
 
+  template <int dims = Dimensions>
+  typename detail::enable_if_t<(dims == 1), size_t>
+  get_local_linear_id_impl() const {
+    id<Dimensions> localId = get_local_id();
+    return localId[0];
+  }
+
+  template <int dims = Dimensions>
+  typename detail::enable_if_t<(dims == 2), size_t>
+  get_local_linear_id_impl() const {
+    id<Dimensions> localId = get_local_id();
+    return localId[0] * groupRange[1] + localId[1];
+  }
```

Comment:
I think we move the whole implementation under `#ifdef __SYCL_DEVICE_ONLY__` and there should be no issues with using C++ 17 features there. Right?

Reply:
But what about host device? `get_local_linear_id` temporarily is not available there (since it requires ABI breaking change in `get_local_id`), in the future it will be enabled. But `get_local_linear_range` and `get_group_linear_range` (which also have 3 specializations) work on host device and we do not want to disable them on host

Reply:
If we can't put it under `#ifdef __SYCL_DEVICE_ONLY__`, let's keep it as is.